### DISCUSSION TITLE
Newsletter about future events

### DISCUSSION THREAD

**Discussion by f14m07** [liedekef/events-made-easy#225] [General]
I've set the newsletter to list future events for the next 7 days. On my local site this works very well but on the production site the limit of the next 7 days is not respected and all future events appear in the newsletter. I'd like to know in which table the time period taken into account is saved. Any ideas to help me understand are welcome.

**Comment by f14m07** [comment]
I have found the table wp_options field option_name eme_cron_new_events_days. The settings are exactely the same in local and production site.

**Comment by liedekef** [comment]
I'm guessing the version is different and it seems I've created a bug :-)
Can you check if this change helps in prod:
https://github.com/liedekef/events-made-easy/commit/8ceed26bfd9f9977a452c569a93317f9b51adfc1

I'm currently using some hackish method that is comparable with named arguments in PHP 8 (but not everyone has that already), and in the latest version I switched one call in eme_cron.php to that.

**Reply by liedekef** [reply]
btw: the next version of EME will set php 8 as a minimal requirement and start using named arguments, so I am removing the hackish code too.

**Reply by f14m07** [reply]
I use the same version of EME on the local and production site

**Reply by f14m07** [reply]
Good news. I have tested with the two new files eme_cron and eme_events. It's now ok on the local and production site.
Thanks a lot Franky

**Reply by liedekef** [reply]
Well, I still needed to test that code though (I assume you took the latest code from github), but thanks for the first feedback :-)
However: you'll need to update the other changed files too, since I removed some code in eme-events.php and eme-rsvp.php and others have been updated accordingly too.
